ansible-collection/docs/modules/cluster.md

```diff
@@ -0,0 +1,159 @@
+| operation | string     | yes      |           | Operation to perform                                                                  |
+| uid       | string     | varies   |           | Cluster unique identifier (required for all operations except CREATE and INSPECT_ALL) |
+| ownership | Dictionary | varies   |           | Ownership information                                                                 |
+| labels    | string     | no       |           | Label for the cluster                                                                 |
+
+#### Ownership Parameters
+
+
+| Parameter                        | Type       | Required | Choices                  | Description                                        |
+| ---------------------------------- | ------------ | ---------- | -------------------------- | ---------------------------------------------------- |
+| ownership                        | dictionary | no       |                          | Cluster ownership and access control configuration |
+| ownership.owner                  | string     | no       |                          | Owner of the cluster                               |
+| ownership.groups                 | list       | no       |                          | List of group access configurations                |
+| ownership.groups[].id            | string     | yes      |                          | Group identifier                                   |
+| ownership.groups[].access        | string     | yes      | `Read`, `Write`, `Admin` | Access level for the group                         |
+| ownership.collaborators          | list       | no       |                          | List of collaborator access configurations         |
+| ownership.collaborators[].id     | string     | yes      |                          | Collaborator identifier                            |
+| ownership.collaborators[].access | string     | yes      | `Read`, `Write`, `Admin` | Access level for the collaborator                  |
+| ownership.public                 | dictionary | no       |                          | Public access configuration                        |
+| ownership.public.type            | string     | no       | `Read`, `Write`, `Admin` | Public access level                                |
+
+### Configuration Parameters
+
+
+| Parameter                  | Type       | Required | Default  | Description                                                        |
+| ---------------------------- | ------------ | ---------- | ---------- | -------------------------------------------------------------------- |
+| cloud_type                 | string     | no       | `OTHERS` | Cloud provider type (`OTHERS`, `AWS`, `AZURE`, `GOOGLE`, `IBM`)    |
+| cloud_credential_ref       | dictionary | varies   |          | Reference to cloud credentials (required if cloud_type is`OTHERS`) |
```

Comment:
cloud_credential_ref is used for AWS, AZURE, GOOGLE and IBM
platform_credential_ref is used for rancher (OTHERS)